Libraries/Opc.Ua.Client/Subscription.cs

```diff
@@ -804,9 +804,10 @@ public bool PublishingStopped
             {
                 lock (m_cache)
                 {
-                    int keepAliveInterval = (int)(Math.Min(m_currentPublishingInterval * m_currentKeepAliveCount, Int32.MaxValue - 500));
+                    int keepAliveInterval = (int)(Math.Min(m_currentPublishingInterval * (m_currentKeepAliveCount + 1), Int32.MaxValue - 500));
+                    TimeSpan timeSinceLastNotification = DateTime.UtcNow - m_lastNotificationTime;
```

Comment:
This calculation depends on System time changes, which on certain systems (VM 's with oversubscribed CPU's or regular system clock changes) can produce inaccurate results. Probably should consider using a monotonic clock for m_lastNotificationTime. Not sure how much this change impacts other code though.

Reply:
good point, I think there is an issue open when the system time changes.
In this case I realized with some commercial PLC that they send the subscription keep alive for e.g. '5' after '5+1' publish interval cycles. Thats why I added the additional margin.